### ISSUE TITLE
getParameters() for Address headers does not return Paramater Entries in correct Order

### ISSUE THREAD

**Issue by gkumar78** [RestComm/sip-servlets#162]
As per the Parameterable API of Sip Servlets 1.1 specs (https://github.com/RestComm/sip-servlets/blob/master/sip-servlets-spec/src/main/java/javax/servlet/sip/Parameterable.java), the getParameters() method should return the collection with elements in "the order of appearance of the parameters in the Parameterable".

As per my understanding of this, if an address header has value such as

"Display name" <sip:username@domain.com>;param1=value1;param2;param3=value3

Then, an iterator over the Set<Map.Entry<String,String>> returned by getParameters() call should return elements in order of appearance in header value. So, param1 should be first element, param2 second and param3 last element returned by iterator next() methods.

However, when I used this API in Mobicents to parse a Contact header value, I found that the order of parameters returned by this method is not same as in header value. Looking at the ParameterableImpl class as well, I found that it uses standard HashMap for keeping parameters which won't be able to handle the order. Instead something like LinkedHashMap needs to be used to maintain the order of parameters.

**Comment by deruelle** [comment]
We could also have a Comparator for that Map. Would you like to try to contribute a fix for that ?

**Comment by gkumar78** [comment]
Yeah sure. I will give it a try next week to implement the fix unless it takes more than couple of hours.
